packages/tocco-ui/src/EditableValue/typeEditors/NumberEdit.js

```diff
@@ -1,37 +1,95 @@
 import PropTypes from 'prop-types'
+import {intlShape} from 'react-intl'
 import React from 'react'
+import NumberFormat from 'react-number-format'
 
-const convertStringToNumber = stringValue => (
-  !stringValue || isNaN(stringValue) ? null : Number(stringValue)
-)
+import {parseLocalePlaceholder, convertStringToNumber} from '../utils'
+
+export const limitValue = maxValueObject => values => {
+  const {formattedValue, floatValue} = values
+  return formattedValue === '' || floatValue <= maxValueObject
+}
+
+export const calculateMaxValue = (prePointDigits, postPointDigits) => {
+  const maxValueMinuend = (10 ** prePointDigits)
+  const maxValueSubtrahend = 1 / (10 ** postPointDigits)
+  return maxValueMinuend - maxValueSubtrahend
+}
+
+export const isAllowedIntegerValue = allowedIntegerObject => values => {
+  const {formattedValue, floatValue} = values
+  const {minValue, maxValue} = allowedIntegerObject
+  return formattedValue === '' || (floatValue >= minValue && floatValue <= maxValue)
+}
+
+export const returnAllowedIntegerObject = (minValue, maxValue) => {
+  return {minValue, maxValue}
+}
 
 const NumberEdit = props => {
+  const {thousandSeparator, decimalSeparator} = parseLocalePlaceholder(props.options.intl.locale)
   const value = props.value === null ? '' : props.value
-  const handleChange = e => {
+
+  const prePointDigits = props.options.prePointDigits
+  const postPointDigits = props.options.postPointDigits
+  const prePointMarker = prePointDigits === 0 ? true : prePointDigits
```

Comment:
i think there are better solution to check whether prePointDigits is a number >= 0 than this. I thank thats what you are trying to achieve?